internal/consumer/dlq_test.go

```diff
@@ -0,0 +1,110 @@
+package consumer
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/stretchr/testify/suite"
+	"github.com/uber-go/kafka-client/internal/util"
+	"github.com/uber-go/tally"
+	"go.uber.org/zap"
+)
+
+type DLQTestSuite struct {
+	suite.Suite
+	saramaProducer *mockSaramaProducer
+	dlq            *bufferedDLQImpl
+	timeLimit      chan time.Time
+	sizeLimit      chan time.Time
+}
+
+func TestDLQTestSuite(t *testing.T) {
+	suite.Run(t, new(DLQTestSuite))
+}
+
+func (s *DLQTestSuite) SetupTest() {
+	s.timeLimit = make(chan time.Time)
+	s.sizeLimit = make(chan time.Time)
+	s.saramaProducer = newMockSaramaProducer()
+	logger := zap.NewNop()
+	s.dlq = &bufferedDLQImpl{
+		topic:            "topic",
+		producer:         s.saramaProducer,
+		timeLimit:        s.timeLimit,
+		sizeLimit:        s.sizeLimit,
+		stopC:            make(chan struct{}),
+		closedC:          make(chan struct{}),
+		messageBufferMap: newMessageBufferMap(),
+		errorMap:         newErrorMap(),
+		scope:            tally.NoopScope,
+		logger:           logger,
+		lifecycle:        util.NewRunLifecycle("dlqTest", logger),
+	}
+}
+
+func (s *DLQTestSuite) TestBatchProducer() {
```

Comment:
can you also include Start()/Stop() in your test case to verify shutdown doesn't cause any deadlocks / goroutine leaks etc i.e.
```dlq.Start() -> dlq.Add().. dlq.Add() -> dlq.Stop```

Reply:
Added a test.